fe/fe-core/src/main/java/com/starrocks/sql/analyzer/SetStmtAnalyzer.java

```diff
@@ -64,22 +64,55 @@
 import org.apache.commons.lang3.StringUtils;
 
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
 
 public class SetStmtAnalyzer {
     public static void analyze(SetStmt setStmt, ConnectContext session) {
         List<SetListItem> setVars = setStmt.getSetListItems();
-        for (SetListItem var : setVars) {
-            if (var instanceof SystemVariable) {
-                analyzeSystemVariable((SystemVariable) var);
-            } else if (var instanceof UserVariable) {
-                analyzeUserVariable((UserVariable) var);
-            } else if (var instanceof SetUserPropertyVar) {
-                analyzeSetUserPropertyVar((SetUserPropertyVar) var);
-            } else if (var instanceof SetNamesVar) {
-                analyzeSetNames((SetNamesVar) var);
-            } else if (var instanceof SetPassVar) {
-                analyzeSetPassVar((SetPassVar) var, session);
+        HashMap<String, UserVariable> cloneUserVars = new HashMap<>();
+        boolean hasUserVar = setVars.stream().anyMatch(var -> var instanceof UserVariable);
+        //Record the order in which user variables appear in SQL
+        //set @a = 1, @b = @c+1, @c = 1; var b shouldn't be reanalyzed.
```

Comment:
```suggestion
        // Record the order in which user variables appear in SQL
        // set @a = 1, @b = @c+1, @c = 1; var b shouldn't be reanalyzed.
```
Better to give a space after `//`